src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/RestHealthCheckService.cs

```diff
@@ -63,32 +61,31 @@ public Task StopAsync(CancellationToken _)
 
         private async Task CheckEndpointHealthAsync()
         {
-            await Task.WhenAll(_serviceEndpointManager.GetEndpoints(_hubName).Select(async endpoint =>
+            if (_serviceEndpointManager.Endpoints.Count > 1 || _enabledForSingleEndpoint)
             {
-                var retry = 0;
-                var isHealthy = false;
-                bool needRetry;
-                do
+                await Task.WhenAll(_serviceEndpointManager.GetEndpoints(_hubName).Select(async endpoint =>
                 {
-                    try
-                    {
-                        using var client = _clientFactory.Create(endpoint);
-                        isHealthy = await client.IsServiceHealthy(default);
-                    }
-                    catch (Exception ex)
+                    var retry = 0;
+                    var isHealthy = false;
+                    bool needRetry;
+                    do
                     {
-                        //Hard to tell if it is transient error, retry anyway.
-                        Log.RestHealthCheckFailed(_logger, endpoint.Endpoint, ex, retry == MaxRetries, _retryInterval);
-                    }
-                    needRetry = !isHealthy && retry < MaxRetries;
-                    if (needRetry)
+                        isHealthy = await IsServiceHealthy(endpoint);
+                        needRetry = !isHealthy && retry < MaxRetries;
+                        if (needRetry)
+                        {
+                            Log.WillRetryHealthCheck(_logger, endpoint.Endpoint, _retryInterval);
+                            await Task.Delay(_retryInterval);
+                        }
+                        retry++;
+                    } while (needRetry);
+                    endpoint.Online = isHealthy;
```

Comment:
what happens when the endpoint count hotreloads to 1 while the last check is Online=false? It does not check again so even if the endpoint is online it does not change back to online?

Reply:
Now the checking will continue if it detects any of the endpoint is offline.